imap/http_dav.c

```diff
@@ -451,6 +451,31 @@ static const struct precond_t {
     { "max-resource-size", NS_CARDDAV },
 };
 
+void dav_precond_as_string(struct buf *buf, struct error_t *err)
+{
+    const struct precond_t *precond = &preconds[err->precond];
+
+    switch (precond->ns) {
+    case NS_DAV:
+        buf_appendcstr(buf, "DAV:");
+        break;
+    case NS_CALDAV:
+        buf_appendcstr(buf, "CALDAV:");
+        break;
```

Comment:
Note that I've chosen to use these strings as namespaces rather than the actual urns for brevity